### DISCUSSION TITLE
[Handle]: Element Click Intercepted By Dynamic Random Untimely Popups

### DISCUSSION THREAD

**Discussion by heyAfsar** [webdriverio/webdriverio#8904] [Q&A]
Hi,

In our Application there is a dynamic popup element that intercepts clicks anytime it appears on screen and fails the entire test case with an error:
![image](https://user-images.githubusercontent.com/20964475/192075314-e011703f-b084-42c3-a26b-e443de903324.png)

Popup Attributes:
- Popups are regular div/span element that pops in after some time of interacting with the UI (ex- Authentication SSO, Feature Announcements, etc.)
- They can appear at anytime within 2-5 minutes of launching the application
-  The popups will Always have a constant Identifier. ex: ```class= "help4-wrapper"```
- All such popups have a dedicated close button on the top right corner. ex: ex: ```class= "help4-wrapper-close"```

Requirement:
To be able to know and handle when an element click is intercepted by any such predefined popups without failing the entire test. 
I should be able to run a function that performs click on the popup close and then continue with the intercepted step earlier and not fail it entirely.

**Comment by christian-bromann** [comment]
> In our Application there is a dynamic popup element that intercepts clicks anytime it appears on screen and fails the entire test case with an error

If this alert pops up randomly, this would be bad. One approach would be to just overwrite the alert global function to do just nothing if this is not important for the test. Maybe there are ways to disable this behavior for testing purposes.

If it is important to the test, you can try to use the `beforeCommand` hook to check if the popup is existing and handle it. This can tremendously slow down tests though. I would be careful using command hooks.

An other opportunity is using DevTools protocol but then you are limited to this only working in Chrome.
